### PR TITLE
Feature/default focus

### PR DESCRIPTION
**Pull request by cloppingemu**
Addresses #7033

Looks like there has been a regression in functionality when GTK version is updated to >= 4.0.0.

With GTK version >= 4.0.0, the async clipboard functionality (Clipboard::read_text_async) is used to populate the input field in the dialog box(es). As a result, the old method of detecting if the accept button should have focus is no longer reliable because checking if the input field is non-empty can happen before the input field is populated.

__Fix:__
- Callback used to validate the clipboard and populate the input field now also sets focus. Affects TorrentUrlChooserDialog and AddTrackerDialog.
- Old method to check if focus should be on the accept button has been removed to prevent duplication of responsibility.

Notes: Setting default behaviour for GTK dialogs to add torrent from url and add tracker.

### CODE REVIEW

**Review by mikedld — COMMENTED** [review]
This looks fine and fixes the issue, preserving current behavior.

It caught my eye though that #7033 mentions the need to move focus to the Open (or Add in case of trackers) button, which is probably applicable to other dialogs as well. This shouldn't be necessary IMHO, and the proper way is to set a default widget for every such dialog, so that pressing Enter/Return accepts it regardless of the currently focused widget (i.e. not requiring users to move focus), which means (for each of those dialogs):
* setting `can-default` property to `True` (for GTK 3 only) on the button we want to activate by default [and maybe other widgets with `receives-default` property enabled if they'll behave inappropriately]
* setting `activates-default` property to `True` (GTK 3) or `1` (GTK 4) on `GtkEntry` widgets
* calling `set_default()` (GTK 3) or calling `set_default_widget()` or setting `default-widget` property (GTK 4) on a dialog with the button we want to activate by default.

Unless you want to do it here (even if just for the two dialogs you're already fixing), I'm okay with merging this PR (the only comment I left applies either way).

File: gtk/OptionsDialog.cc [hunk 1]
```diff
@@ -427,16 +427,19 @@ TorrentUrlChooserDialog::TorrentUrlChooserDialog(
...
     if (e->get_text_length() == 0)
     {
         e->grab_focus();
     }
     else
     {
-        get_widget_for_response(TR_GTK_RESPONSE_TYPE(ACCEPT))->grab_focus();
+        accept->grab_focus();
     }
+#endif
```
[the remark this answers is not in the document]
Comment (cloppingemu) [reply gtk/OptionsDialog.cc:444]: True, and it should work for GTK pre and post v4.0.0. Removing section and committing again.

**Comment by cloppingemu** [comment]
Happy to have a look at the default widget approach to accepting the dialog box(es) here. Just to clarify, is the hope that you should be able to hit Return when GtkEntry is in focus and be able to accept the dialog box (pretty sensible behavior)?

**Comment by mikedld** [comment]
> is the hope that you should be able to hit Return when GtkEntry is in focus and be able to accept the dialog box

Yes. Hitting Enter/Return when Cancel button is focused is supposed to cancel though.

**Comment by cloppingemu** [comment]
Using the activates-default property to add a url when enter is pressed was rather straight-forward. It appears to be functioning as intended locally with GTK4.
Have not tested with GTK3 because I'm not sure how to build with GTK3.
Previous change where the focus is grabbed if the url in the clipboard is valid has been removed.

Think I've uncovered a bug or two when adding a URL (also exists in main). I'm still investigating the source but I'll create an issue for it so it is not just rolled into this PR.

**Comment by cloppingemu** [comment]
Turns out I was missing a property in the GTK3 buttons. GTK3 application now also add torrents and trackers when enter is pressed from url entry.

**Comment by mikedld** [comment]
Also please add "Notes:" to the PR description as other PRs do, it'll go into the release notes.

File: gtk/ui/gtk4/TorrentUrlChooserDialog.ui [hunk 2]
```diff
@@ -24,6 +24,7 @@
                   </object>
                 </child>
                 <child>
+
```
[the remark this answers is not in the document]
Comment (cloppingemu) [reply gtk/ui/gtk4/TorrentUrlChooserDialog.ui]: Unintended change indeed. Removed

**Comment by cloppingemu** [comment]
Hi, I don't think I follow what you mean by

> Also please add "Notes:" to the PR description as other PRs do

I see some PRs have the "notes" label but I don't think I can add labels to this PR.

**Comment by mikedld** [comment]
I'm talking about PR description (i.e. text you have access to), you can search for "notes" in closed/merged PRs to see examples of it.

**Comment by cloppingemu** [comment]
Added notes to the description.

**Review by mikedld — APPROVED** [review]

**Pull request merged**